map/views.py

```diff
@@ -7,23 +10,63 @@ def mapview(request):
     craigslist_datas = CraigslistLocation.objects.all()
 
     for data in craigslist_datas:
-        title = data.name
-        url = '<a href="' + data.url + '" target="_blank">' + data.url + "</a>"
-        description = data.price + "\n" + str(data.date_time) + "\n" + url
+        if not CityStreetSpot.objects.filter(c_id=data.c_id).exists():
+            title = data.name
+            url = '<a href="' + data.url + '" target="_blank">' + data.url + "</a>"
+            description = data.price + "\n" + str(data.date_time) + "\n" + url
+
+            if data.has_image:
+                image_url = get_img_url(data.url)
```

Comment:
Would it make more sense to load the image when we retrieve the CraigslistLocatin instead of here when we're trying to load the map?

Reply:
Yes, it makes more sense. Will do that next sprint! Thanks!